PDBminer

```diff
@@ -73,30 +73,94 @@ parser.add_argument("-f", "--snakefile",
 args = parser.parse_args()
 path = os.getcwd()
 
-def check_input_file(df):
+def uniprot_accession_number_type(df):
+    """
+    This function takes the input to PDBminer read as a pandas dataframe 
+    and check if the uniprot accession numbers are correctly formatted.
+
+    Parameters
+    ----------
+    df : The input dataframe generated from the input CSV.
 
+    Returns
+    -------
+    problems : If the uniprot accession number is not a string, the format
+    is wrong and will return a list of string(s) with the offending input.
+
+    """
     problems = []
     
-    for i in range(len(df)):    
+    #Uniprot Accession number check:
+    problematic_uniprots = df.uniprot[~df.uniprot.apply(lambda x: isinstance(x, str))]
+    for uniprot_value in problematic_uniprots:
+        problems.append(f"there is a problem with the uniprot input: {uniprot_value}")
+    
+    # If there are issues with the uniprot input, the program should end, 
+    # since this is the only mandatory element of the input.
+    if problems != []:
+        return problems
+    
+def retrieve_hugo_name(uniprot_accession_number):
+    """
+    This functon takes a uniprot accession number and retrieves a gene name
+    from the uniprot database. If Uniprot DB is down, the program will
+    end with an error.
+
+    Parameters
+    ----------
+    uniprot_accession_number : Accession number. 
+
+    Returns
+    -------
+    hugo_name : The listed gene name for the uniprot accession number.
+
+    """
+    
+    #try to get the hugoname from the uniprot database. 
+    # If the uniprot datbase is down, the program will exit.
+    try: 
+        response = requests.get(f"https://www.uniprot.org/uniprot/{uniprot_accession_number}.json")
+    except ConnectionError as e:
+        with open("log.txt", "a") as textfile:
+            textfile.write(f"EXITING: Uniprot database API controlled and rejected for {uniprot_accession_number} when finding the Hugo Name, connection error.\n")
```

Comment:
"controlled and rejected" makes it feel a bit like we got our password wrong or something, can we just say we couldn't connect to the endpoint?